packages/mobile/src/components/fields/PriceField.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect, useCallback } from 'react'
+
+import {
+  decimalIntegerFromHumanReadable,
+  decimalIntegerToHumanReadable,
+  filterDecimalString,
+  padDecimalValue
+} from '@audius/common'
+import { useField } from 'formik'
+import type {
+  NativeSyntheticEvent,
+  TextInputChangeEventData,
+  TextInputFocusEventData
+} from 'react-native'
+
+import { Text } from '../core/Text'
+
+import type { TextFieldProps } from './TextField'
+import { TextField } from './TextField'
+
+const messages = {
+  dollars: '$'
+}
+
+/** Implements a Formik field for entering a price, including default dollar sign
+ * adornment and conversion logic to/from human readable price. Internal value is stored
+ * as an integer number of cents.
+ */
+export const PriceField = (props: TextFieldProps) => {
+  const [{ value }, , { setValue: setPrice }] = useField<number>(props.name)
```

Comment:
Exactly. We want to keep the value as an integer, but show the string with decimal point to the user.
So we keep separate copies of the two and:
* When the humanized value changes, we parse it and convert it to an integer (using the utils)
* When the field blurs, we pad the user-input value to a fixed number of decimal places

And yeah, the renaming of `setValue` to `setPrice` is mostly personal style because I generally prefer that you know _what_ value you are setting when you read the code. In this case, it's probably unnecessary because the control only handles a single value and it's clear what we're operating on.